packages/fuel-indexer-schema/src/lib.rs

```diff
@@ -211,6 +214,52 @@ impl FtColumn {
                 Some(val) => format!("'{val}'"),
                 None => String::from(NULL_VALUE),
             },
+            FtColumn::List(value) => {
+                match value {
+                    Some(list) => {
+                        let discriminant = std::mem::discriminant(&list[0]);
+                        let type_id =
+                            i32::from(ColumnType::from(list[0].to_string().as_str()));
+                        let type_bytes = hex::encode(type_id.to_be_bytes());
+                        let elems = list
+                            .iter()
+                            .map(|e| {
+                                if std::mem::discriminant(e) != discriminant {
+                                    panic!(
+                                        "List elements are not of the same column type; expected: {:#?}, actual: {:#?}",
+                                        discriminant,
+                                        std::mem::discriminant(e)
+                                    )
+                                } else {
+                                    // TODO: Should each element be encoded to bytes on its own?
+                                    // How will we successfully store Some() and None for the same type
+                                    // in one list?
+                                    e.query_fragment()
+                                }
+                            })
+                            .collect::<Vec<String>>();
+                        format!(
+                            "'{}{}'",
+                            type_bytes,
+                            hex::encode(elems.join(", ").as_bytes())
```

Comment:
- I think you've pretty much got it here
- However note building each frag `e.query_fragment()`  separately is gonna make thing final result a little wonky `"'0x1203', '0x01233', '0x33294'"`
- I think it might be better to convert each individual item into bytes, concat the type id, then hex the final result:


```rust
let mut type_id = 123i32.to_le_bytes().to_vec();
let parts: Vec<u8> = list.iter().map(|e| e.into()).collect::<Vec<Vec<u8>>>().concat();
let final = type_id.concat(&parts);
let result = hex::encode(final);
```

☝🏽 pseudo-code

Reply:
I agree with that; it definitely feels better overall.